### PR TITLE
Enable Nomad ACL integration with Vault

### PR DESCRIPTION
**Pull request by lawliet89**

### CODE REVIEW

File: modules/core/README.md [hunk 1]
```diff
@@ -232,11 +232,34 @@ After you have initialised and unsealed Vault, you can use the
...
+
 The default `user_data` scripts for Nomad servers and clients will automatically detect that the
 policies have been setup and will configure themselves correctly. To update your cluster to use
 the new Vault integration, simply follow the section below to update the Nomad servers first and
 then the clients.
 
+### Nomad ACL
+
+[ACL](https://www.nomadproject.io/guides/acl.html) can be enabled for Nomad so that only users
+with the necessary tokens can submit jobs. This module only enables to built in access controls
```
Comment (guangie88) [inline modules/core/README.md]: `This module only enables to built in access controls`, is it `This module only enables built-in access controls`?

File: modules/nomad-acl/README.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
+# Nomad ACL
+
+[ACL](https://www.nomadproject.io/guides/acl.html) can be enabled for Nomad so that only users
+with the necessary tokens can submit jobs. This module only enables to built in access controls
```
Comment (guangie88) [inline modules/nomad-acl/README.md]: Same issue here with the text.

File: modules/nomad-acl/README.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+consul kv get \
+    -http-addr https://consul.example.com \
+    terraform/nomad-acl/enabled
+```
+
+Alternatively, you can use Terraform too:
+
+```hcl
+resource "consul_keys" "bootstrap" {
+    key {p
```
Comment (chrissng) [inline modules/nomad-acl/README.md]: nitpick: remove the typo here

File: modules/core/packer/roles/nomad/files/configure.sh [hunk 4]
```diff
@@ -21,10 +21,11 @@ function print_usage {
   echo
   echo -e "  --server\t\tIf set, configure in server mode. Optional. Exactly one of --server or --client must be set."
   echo -e "  --client\t\tIf set, configure in client mode. Optional. Exactly one of --server or --client must be set."
-  echo -e "  --config-path\t\tThe path to write the config file to. Optional. Default is the absolute path of '../config/vault.hcl', relative to this script."
+  echo -e "  --config-dirh\t\tThe path to write the config files to. Optional. Default is the absolute path of '../config', relative to this script."
```
Comment (guangie88) [inline modules/core/packer/roles/nomad/files/configure.sh]: `--config-dirh` -> `--config-dir`

File: modules/core/packer/roles/nomad/files/configure.sh [hunk 5]
```diff
@@ -21,10 +21,11 @@ function print_usage {
...
   echo -e "  --vault-service\t\tName of Vault service to query in Consul. Optional. Defaults to 'vault'."
   echo -e "  --vault-port\t\tPort of Vault service. Optional. Defaults to '8200'."
-  echo -e "  --consul-prefix\t\tPath prefix in Consul KV store to query for Vault configuration status. Optional. Defaults to terraform/nomad-vault-integration/"
+  echo -e "  --consul-prefix\t\tPath prefix in Consul KV store to query for integration status. Optional. Defaults to terraform/"
+  echo -e "  --user\t\tThe user to run Nomad as. Optional. Default is to use the owner of --config-dirh."
```
Comment (guangie88) [inline modules/core/packer/roles/nomad/files/configure.sh]: Same `--config-dirh`.

File: modules/nomad-acl/README.md [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+Make sure you have properly configured Vault with the appropriate
+[authentication methods](https://www.vaultproject.io/docs/auth/index.html) so that your users can
+authenticate with Vault to get the necessary tokens and credentials.
+
+### Bootstrapping Nomad ACL
+
+We have to manually [bootstrap](https://www.nomadproject.io/guides/acl.html#bootstrapping-acls) ACLs
+on the Nomad servers first.
+
+Firstly, this requires that the Nomad servers and servers are configured with the
```
Comment (guangie88) [inline modules/nomad-acl/README.md]: Suggestion: `this requires that the Nomad servers and servers are configured with the` -> `this requires the Nomad clients and servers to be configured with`.

File: modules/nomad-acl/README.md [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+```
+
+After this is done, update the Nomad servers _first_ according to the instructions provided
+in the Core module. You might get `permission denied` if you try to use the `nomad server members`
+command to check if the servers are up.
+
+You can use the [status API](https://www.nomadproject.io/api/status.html) instead which does not
+require any ACL.
+
+Secondly, we will need to generate a Nomad Boostrap token. Use the
```
Comment (guangie88) [inline modules/nomad-acl/README.md]: `Boostrap` -> `Bootstrap`

**Pull request merged**